tests/test_str.py

```diff
@@ -12,7 +13,6 @@
 
 
 def test_Format():
-    # TODO Ammar: The invariants that we test here should be visible in the documentation of str.Format.
```

Comment:
You did not address this TODO. Invariants tested here are for example: In which order are the arguments given to str.format? What happens if I supply the same key twice, once in _kwargs, once in kwargs?